### DISCUSSION TITLE
Need to assign contactRole to a contact

### DISCUSSION THREAD

**Discussion by dpreiss** [netsuitephp/netsuite-php#113] [Q&A] [answered]
I've tried several ways to assign a contactRole to a contact.  The internal id value of the Contact Role is -10.

I tried setting the contactrole when I create the contact, but it did not work, code:

`$contact = new Contact();
        $contact->firstName = $data['firstname'];
        $contact->lastName = $data['lastname'];
        $contact->phone = $data['phone'];
        $contact->email = $data['email'];
        $contact->contactrole = -10; //primary contact
        $contact->company = new RecordRef();
        $contact->company->internalId = 1028

        $request = new AddRequest();
        $request->record = $contact;
        $addContact = $service->add($request);`

I also tried the following code using ContactAccessRolesList to assign the role to the contact associated with the customer:

`$role = new ContactAccessRoles();
$role->contact = new RecordRef();
$role->contact->internalId = 55811;
$role->role = new RecordRef();
$role->role->internalId = "-10";  //I tried with and without the quotes, but same error invalid role reference key -10
$role->role->type = "contactRole";

$contactRolesList = new ContactAccessRolesList();
$contactRolesList->contactRoles = $role;

$customer = new Customer();
$customer->internalId = 32258;
$customer->contactRolesList = $contactRolesList;

$request = new UpdateRequest();
$request->record = $customer;
$response = $service->update($request);`

The result of the above code is:  Invalid role reference key -10

I know the internal id is -10.   Any clues on how to assign the contact role to the contact via api?

**Comment by dpreiss** [comment, marked as answer]
After posting this, I found the answer.   In case someone needs to do this, here is the solution:

`$role = new AttachContactReference();
$role->attachTo = new RecordRef();
$role->attachTo->type = "customer";
$role->attachTo->internalId  = 32258;   //customer internal id
$role->contact = new RecordRef();
$role->contact->internalId = 55811; //contact internal id
$role->contactRole = new RecordRef();
$role->contactRole->internalId = "-10"; //contact role internal id

$request = new AttachRequest();
$request->attachReference = $role;
$response = $service->attach($request);`

**Reply by neclimdul** [reply]
Sometimes you have to explain the problem to find the solution 😃🦆
